clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp

```diff
@@ -26,6 +26,17 @@ void tooling::dependencies::configureInvocationForCaching(
   auto &FrontendOpts = CI.getFrontendOpts();
   FrontendOpts.CacheCompileJob = true;
   FrontendOpts.IncludeTimestamps = false;
+  if (CI.getFrontendOpts().ProgramAction != frontend::ActionKind::EmitObj) {
+    auto &CodeGenOpts = CI.getCodeGenOpts();
```

Comment:
Nit: you could hoist this and change the other uses of `CI.getCodeGenOpts()` below